ci/bamboo/build-osx.sh

```diff
@@ -1,32 +1,40 @@
 #!/bin/bash
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have purchased from
+# Numenta, Inc. a separate commercial license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+#
+# OPTIONS:
+#  --user: If provided, will pass along the --user argument to any pip install
```

Comment:
@rhyolight, why do you prefer to pass "--user" from the outside? Shouldn't this build script abstract within it any knowledge/logic that is specific to the OSX build? Here is where I was trying to go with this: Think of it as our Bamboo OSX adapter on top of `build-and-test-nupic-bindings.sh`, so nothing outside should ever need to pass `--user` to this script. I.e., making the OSX build using this script in our Bamboo environment won't work without `--user`, so it shouldn't be a command-line arg at all. Instead, it should explicitly and unconditionally pass `--user` to `build-and-test-nupic-bindings.sh`